dso-l2/src/main/java/com/tc/objectserver/handler/ProcessTransactionHandler.java

```diff
@@ -193,7 +196,12 @@ private synchronized void addMessage(ClientID sourceNodeID, EntityDescriptor des
       }
     } else {
       // If there was an exception of any sort, just pass it back as a failure.
+      logger.error("caught exception while handling the request ", uncaughtException);
       serverEntityRequest.failure(uncaughtException);
+      // crash the server if exception is a EntityUserException
+      if(uncaughtException instanceof EntityUserException) {
```

Comment:
Why are we checking the exception type?  Can we throw when we first create/catch it, instead of at this later point?